### PR TITLE
[Enhancement] Fix CTAS failure caused by HMS getAllTable

### PR DESCRIPTION
**Pull request by zenoyang**
Why I'm doing:
When CTAS is used to create Hive external tables, if there are a large number of tables in the db, invoking HMS `getAllTables` will time out and fail, and HMS may have bad effects.

What I'm doing:
Avoid `listTableNames` by using `getTable` to determine whether a table exists.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java [hunk 1]
```diff
@@ -243,7 +243,8 @@ public boolean createTable(CreateTableStmt stmt) throws DdlException {
                     ErrorReport.reportDdlException(ErrorCode.ERR_BAD_DB_ERROR, dbName);
                 }
 
-                if (listTableNames(catalogName, dbName).contains(tableName)) {
+                Table table = connectorMetadata.get().getTable(dbName, tableName);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:246]: could we catch the exception to check if table exists?

Comment (zenoyang) [reply fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:246]: The exception has been caught in the `HiveMetadata#getTable` function. If caught, `null` will be returned.

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/CachingHiveMetastore.java [hunk 2]
```diff
@@ -292,6 +292,10 @@ public Table getTable(String dbName, String tableName) {
         return get(tableCache, HiveTableName.of(dbName, tableName));
     }
 
+    public boolean tableExists(String dbName, String tableName) {
+        return metastore.tableExists(dbName, tableName);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/CachingHiveMetastore.java:296]: there are two level caches. catalog level cache and query level cache.

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java [hunk 3]
```diff
@@ -112,6 +112,10 @@ default Table getTable(String dbName, String tblName) {
         return null;
     }
 
+    default boolean tableExists(String dbName, String tblName) {
+        return getTable(dbName, tblName) != null;
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:116]: it's best to use `listPartitionNames.contains` to check as default.

Comment (zenoyang) [reply fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java]: Done

**Review by stephen-shelby — DISMISSED** [review]

**Review by Youngwb — DISMISSED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**